File: backend/app/schemas/event.py
from typing import Literal

from pydantic import BaseModel, Field


EventStatus = Literal["active", "monitoring", "resolved"]
EventPriority = Literal["critical", "high", "medium", "low"]
EventCause = Literal[
    "Vehicle Breakdown",
    "Accident",
    "Construction",
    "Water Logging",
    "Tree Fall",
    "Signal Failure",
    "Protest",
    "VIP Movement",
]
VehicleType = Literal["Truck", "Car", "Bus", "Two-Wheeler", "Auto", "None"]


class TrafficEvent(BaseModel):
    id: str
    code: str
    cause: EventCause
    status: EventStatus
    priority: EventPriority
    zone: str
    lat: float
    lng: float
    description: str
    vehicleType: VehicleType
    createdAt: str
    eta: str
    closureRisk: float = Field(ge=0, le=1)
    hotspotRisk: float = Field(ge=0, le=1)
    affectedRadius: int = Field(ge=0)
    reportedBy: str
    recommendedAction: str

File: frontend/src/types/event.ts
export type EventStatus = "active" | "monitoring" | "resolved";
export type EventPriority = "critical" | "high" | "medium" | "low";
export type EventCause =
  | "Vehicle Breakdown"
  | "Accident"
  | "Construction"
  | "Water Logging"
  | "Tree Fall"
  | "Signal Failure"
  | "Protest"
  | "VIP Movement";
export type VehicleType = "Truck" | "Car" | "Bus" | "Two-Wheeler" | "Auto" | "None";

/** Full traffic event record from GET /api/events */
export interface Event {
  id: string;
  code: string;
  cause: EventCause;
  status: EventStatus;
  priority: EventPriority;
  zone: string;
  lat: number;
  lng: number;
  description: string;
  vehicleType: VehicleType;
  createdAt: string;
  eta: string;
  closureRisk: number;
  hotspotRisk: number;
  affectedRadius: number;
  reportedBy: string;
  recommendedAction: string;
}

/** Map-layer event from GET /api/map-events (same shape as Event for map rendering) */
export type MapEvent = Event;

/** @deprecated Use Event — kept for existing imports */
export type TrafficEvent = Event;

export interface EventsFilter {
  priority?: string | null;
  cause?: string | null;
  query?: string | null;
}
